src/doc-editors/DynamicRulesEditor.vue

```diff
@@ -0,0 +1,378 @@
+                      </td>
+                    </tr>
+                    </tbody>
+                  </table>
+                </div>
+              </div>
+            </div>
+          </div>
+        </div>
+      </div>
+    </div>
+  </section>
+</template>
+<script lang="ts">
+// @go-to-route="emitGoToRoute"
+import _ from 'lodash'
+import {defineComponent} from 'vue'
+import TagAutocompleteInput from '@/components/TagAutocompleteInput.vue'
+import {
+  CustomResponse,
+  Dictionary,
+  DynamicRule,
+  DynamicRuleTargetOptionType,
+  GlobalFilter,
+  IncludeExcludeType,
+} from '@/types'
+import DatasetsUtils from '@/assets/DatasetsUtils'
+import RequestsUtils from '@/assets/RequestsUtils'
+import {AxiosResponse} from 'axios'
+
+
+export default defineComponent({
+  name: 'DynamicRulesEditor',
+  props: {
+    selectedDoc: Object,
+    selectedDocMatchingGlobalFilter: Object,
+    selectedBranch: String,
+    apiPath: String,
+  },
+  components: {
+    TagAutocompleteInput,
+  },
+  data() {
+    const dynamicRuleTargets = DatasetsUtils.dynamicRuleTargets
+    return {
+      dynamicRuleTargets,
+      filters: ['include', 'exclude'] as IncludeExcludeType[],
+      titles: DatasetsUtils.titles,
+      addNewTagColName: null,
+      removable: false,
+      options: ['remote_addr', 'organization', 'cookie', 'geoip_city_country_name', 'planet', 'request_headers',
+        'request_body'] as DynamicRuleTargetOptionType[],
+      customResponseNames: [] as [CustomResponse['id'], CustomResponse['name']][],
+    }
+  },
+  computed: {
+    localDoc(): DynamicRule {
+      if (!this.selectedDoc) {
+        console.log('selectedDoc is Empty')
+      }
+      return _.cloneDeep(this.selectedDoc as DynamicRule)
+    },
+    localGlobalFilterDoc(): GlobalFilter {
+      if (!this.selectedDocMatchingGlobalFilter) {
+        console.log('selectedDocMatchingGlobalFilter is Empty')
+      }
+      return _.cloneDeep(this.selectedDocMatchingGlobalFilter as GlobalFilter)
+    },
+    duplicateTags(): Dictionary<string> {
+      const doc = this.localDoc
+      const allTags = _.concat(doc['include'], doc['exclude'])
+      const dupTags = _.filter(allTags, (val, i, iteratee) => _.includes(iteratee, val, i + 1))
+      return _.fromPairs(_.zip(dupTags, dupTags))
+    },
+    selectedDocTags: {
+      get: function(): string {
+        if (this.localGlobalFilterDoc.tags && this.localGlobalFilterDoc.tags.length > 0) {
+          return this.localGlobalFilterDoc.tags.join(' ')
+        }
+        return ''
+      },
+      set: function(tags: string): void {
+        this.localGlobalFilterDoc.tags = tags.length > 0 ? _.map(tags.split(' '), (tag) => {
+          return tag.trim()
+        }) : []
+        this.emitMatchDoc()
+      },
+    },
+  },
+  emits: ['update:selectedDoc', 'update:selectedDocMatchingGlobalFilter'],
+  methods: {
+    emitDocUpdate() {
+      this.$emit('update:selectedDoc', this.localDoc)
+    },
+    emitMatchDoc() {
```

Comment:
Let's add the `update` to the name

Perhaps:

```suggestion
    emitDocMatchUpdate() {
```